### PR TITLE
build 2.3.0

### PR DESCRIPTION
**Pull request by ocefpaf**
The `gdalwarp` warp is failing (maybe the API changed?)
We are ignoring that failure for now.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
 
 source:
   url: http://download.osgeo.org/gdal/{{ version }}/gdal-{{ version }}.tar.xz
-  sha256: 441eb1d1acb35238ca43a1a0a649493fc91fdcbab231d0747e9d462eea192278
+  sha256: 6f75e49aa30de140525ccb58688667efe3a2d770576feb7fbc91023b7f552aa2
   patches:
     # Fixes compilation problems with JPEG.
     # https://lists.osgeo.org/pipermail/gdal-dev/2016-January/043443.html
-    - disable_jpeg12.patch
+    # - disable_jpeg12.patch
```
Comment (ocefpaf) [inline recipe/meta.yaml]: @gillins if I got the code changes correctly this patch can be dropped.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
     # BUILT_AS_DYNAMIC_LIB.
-    - windowshdf5.patch
+    # - windowshdf5.patch
     # Use multiple cores on Windows.
-    - multiprocessor.patch
+    # - multiprocessor.patch
```
Comment (ocefpaf) [inline recipe/meta.yaml]: These two may need to be refreshed. Can you confirm that @gillins?
(I can do that later over the weekend.)

File: recipe/run_test.sh [hunk 3]
```diff
@@ -3,20 +3,31 @@
...
-proj4="+y_0=2400761.714982585 +lat_ts=-19.6097 +a=6378137.0 +proj=merc +units=m +b=6356752.3142 +lat_0=-19.6097 +x_0=-0.0 +lon_0=-70.7691"
-
-gdalwarp -s_srs "+proj=latlong" -t_srs "$proj4" -of EHdr grid.asc grid.flt
-
-# Test ISIS3/USGS driver `SetNoDataValue()` issue.
+echo ""
+echo "GDAL WARP TEST"
+echo ""
+# proj4="+y_0=2400761.714982585 +lat_ts=-19.6097 +a=6378137.0 +proj=merc +units=m +b=6356752.3142 +lat_0=-19.6097 +x_0=-0.0 +lon_0=-70.7691"
+# gdalwarp -s_srs "+proj=latlong" -t_srs "$proj4" -of EHdr grid.asc grid.flt
```
Comment (ocefpaf) [inline recipe/run_test.sh]: This test is failing but my guess us some API change. I'll take a closer look later today.

**Review by ocefpaf — COMMENTED** [review]
@gillins the Linux and OS X are building again :tada: 

No idea what is going on with Windows. Can you take a quick look?

File: recipe/meta.yaml (hunk 1 above)
Comment (gillins) [reply recipe/meta.yaml]: Yes I think you are right, but you will have to add `JPEG12_SUPPORTED=OFF` or something to `bld.bat`. I think....

You will also have to skip Py27 on Windows.

I'm a bit tied up for the next week or so but will have a proper look after that if it still isn't working.

Comment (gillins) [reply recipe/meta.yaml]: I think you need to change `OPENJPEG_CFLAGS="-I%LIBRARY_INC%" ^` -> `OPENJPEG_CFLAGS="-I%LIBRARY_INC%\openjpeg-2.3" ^` in `bld.bat`. Older versions of `libgdal` appended the `openjpegX.X` part automatically but doesn't seem to happen now....

File: recipe/run_test.sh (hunk 3 above)
Comment (ocefpaf) [reply recipe/run_test.sh]: @mhearne-usgs this test now returns `ERROR 1: Couldn't determine X spacing` do you know if `gdal 2.3.0` changed the API here or is this a real failure?

**Comment by gillins** [comment]
Sorry @ocefpaf I gave you incorrect information. I've made some changes to get it to work....

**Comment by ocefpaf** [comment]
The `gdalwarp` test is failing on Windows too but AppVeyor is not failing :-/

See https://ci.appveyor.com/project/conda-forge/libgdal-feedstock/build/1.0.102/job/fv4xlwreg3gei8mo#L5529

Not sure what happened there, looks like we have an old syntax or something. I'll try to find docs to see if we can fix that. Otherwise I'm inclined to drop that test.

**Comment by gillins** [comment]
I added `if errorlevel 1 exit 1` to each test (bat files don't have a `set -e` option) so it now exits on failure - same as other platforms. 

I can't shed any light on the `gdawarp` failure I'm afraid...

**Comment by kmuehlbauer** [comment]
@ocefpaf @gillins I didn't had time so far to test this. But looking at the appveyor output, I've never seen the double `""` used. Slight chance, that the console is misinterpreting things there?

**Comment by gillins** [comment]
> Slight chance, that the console is misinterpreting things there?

Well spotted! Will see if that helps...

**Comment by ocefpaf** [comment]
I'm inclined to skip the `gdalwarp` test. It was added b/c a user reported an issue that we fixed but it seems to be stale now. What do you think @conda-forge/libgdal team?

**Comment by ocefpaf** [comment]
I don't think we can use this to build `rasterio` and `fiona` yet. I'm following the development of those tools and I'll update the pinning when they are working with `2.3.0`.

**Pull request merged**